packages/enhanced-img/src/preprocessor.js

```diff
@@ -0,0 +1,188 @@
+import MagicString from 'magic-string';
+import { parse } from 'svelte-parse-markup';
+import { walk } from 'svelte/compiler';
+
+const ASSET_PREFIX = '___ASSET___';
+
+// TODO: expose this in vite-imagetools rather than duplicating it
+const OPTIMIZABLE = /^[^?]+\.(avif|heif|gif|jpeg|jpg|png|tiff|webp)(\?.*)?$/;
+
+/**
+ * @returns {import('svelte/types/compiler/preprocess').PreprocessorGroup}
+ */
+export function image() {
+	return {
+		markup({ content, filename }) {
```

Comment:
Can we early-exit if no `<enhanced:img>` is found?

```suggestion
		markup({ content, filename }) {
			if (!content.includes('<enhanced:img')) return;

```